File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = n-len(lost)
    lost.sort()
    reserve.sort()
    l = []
    for r in lost:
        if r in reserve:
            answer +=1
            reserve.remove(r)
            l.append(r)
    for i in l:
        lost.remove(i)
    for i in lost:
        if i-1 in reserve:
            reserve.remove(i-1)
            answer +=1
        elif i+1 in reserve:
            reserve.remove(i+1)
            answer +=1
    return answer

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = n - lost.length;
        HashSet<Integer> set = new HashSet<>();
        List<Integer> lost2 = new ArrayList<>();
        List<Integer> reserve2 = new ArrayList<>();
        for (Integer i: lost) {
            lost2.add(i);
        }
        for (Integer i: reserve) {
            if (lost2.contains(i)) {
                set.add(i);
            } else {
                reserve2.add(i);
            }
        }
        Collections.sort(reserve2);
        
        for (Integer i: reserve2) {
            int c = 0;
            if (lost2.contains(i-1)) {
                if (set.contains(i-1)) {
                } else {
                    set.add(i-1);
                    c +=1;
                }
            }
            if (lost2.contains(i+1) && c == 0) {
                if (set.contains(i+1)) {
                } else {
                    set.add(i+1);
                }
            }
        }
        return answer + set.size();
    }
}